sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -0,0 +1,620 @@
+   *                  Otherwise, it is a Literal value.
+   * @return a BigDecimal value
+   */
+  def numericLiteralToBigDecimal(
+       literal: Any,
+       dataType: DataType,
+       isNumeric: Boolean = false)
+    : BigDecimal = {
+    dataType match {
+      case _: IntegralType =>
+        val stringValue: String =
+          if (isNumeric) literal.toString
+          else literal.asInstanceOf[Literal].value.toString
+        BigDecimal(java.lang.Long.valueOf(stringValue))
+
+      case _: FractionalType =>
+        if (isNumeric) BigDecimal(literal.asInstanceOf[Double])
+        else BigDecimal(literal.asInstanceOf[Literal].value.asInstanceOf[Double])
+
+      case DateType =>
+        if (isNumeric) BigDecimal(literal.asInstanceOf[BigInt])
+        else {
+          val dateLiteral = literal.asInstanceOf[Literal].dataType match {
+            case StringType =>
+              DateTimeUtils.stringToDate(
+              literal.asInstanceOf[Literal].value.asInstanceOf[UTF8String]).
+              getOrElse(0).toString
+            case _ => literal.asInstanceOf[Literal].value.toString
+          }
+          BigDecimal(java.lang.Long.valueOf(dateLiteral))
+        }
+
+      case TimestampType =>
+        if (isNumeric) BigDecimal(literal.asInstanceOf[BigInt])
+        else {
+          val tsLiteral = literal.asInstanceOf[Literal].dataType match {
+            case StringType =>
+              DateTimeUtils.stringToTimestamp(
+                literal.asInstanceOf[Literal].value.asInstanceOf[UTF8String]).
+                getOrElse(0).toString
+            case _ => literal.asInstanceOf[Literal].value.toString
+          }
+          BigDecimal(java.lang.Long.valueOf(tsLiteral))
+        }
+    }
+  }
+
+  /**
+   * Returns a percentage of rows meeting an equality (=) expression.
+   * This method evaluates the equality predicate for all data types.
+   *
+   * @param attrRef an AttributeReference (or a column)
+   * @param literal a literal value (or constant)
+   * @param update a boolean flag to specify if we need to update ColumnStat of a given column
+   *               for subsequent conditions
+   * @return a doube value to show the percentage of rows meeting a given condition
+   */
+  def evaluateEqualTo(
+      attrRef: AttributeReference,
+      literal: Literal,
+      update: Boolean)
+    : Option[Double] = {
+
+    val aColStat = mutableColStats(attrRef.exprId)
+    val ndv = aColStat.distinctCount
+
+    // decide if the value is in [min, max] of the column.
+    // We currently don't store min/max for binary/string type.
+    // Hence, we assume it is in boundary for binary/string type.
+
+    val inBoundary: Boolean = attrRef.dataType match {
+      case _: NumericType | DateType | TimestampType =>
+        val statsRange =
+          Range(aColStat.min, aColStat.max, attrRef.dataType).asInstanceOf[NumericRange]
+        val lit = numericLiteralToBigDecimal(literal, attrRef.dataType)

```

Comment:
just pass literal.value here, then you can remove isNumeric parameter for `numericLiteralToBigDecimal`